app/controllers/publishers_controller.rb

```diff
@@ -110,14 +110,12 @@ def protect
 
   def ensure_email
     token = params[:token]
-    confirm_email = params[:confirm_email]
-
     publisher = Publisher.find(params[:id])
 
-    if PublisherTokenAuthenticator.new(publisher: publisher, token: token, confirm_email: confirm_email).perform(consume: false)
+    if token && PublisherTokenAuthenticator.new(publisher: publisher, token: token).perform(consume: false)
```

Comment:
This fails if token isn't present, and if the token isn't present it can't be valid so we throw an error.